### PR TITLE
FEATURE: Implement refunds from dashboard

### PR DESCRIPTION
**Pull request by justindirose**
A common feature request has been to implement refunds from the admin dashboard. Currently for subscriptions only (not one time purchases). Functions via way of a checkbox in a modal, which then fires the refund request to the backend.

### CODE REVIEW

File: assets/javascripts/discourse/routes/admin-plugins-discourse-subscriptions-subscriptions.js.es6 [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
+      subscription.set("loading", true);
+      subscription
+        .destroy(refund)
+        .then((result) => subscription.set("status", result.status))
+        .catch((data) =>
+          bootbox.alert(data.jqXHR.responseJSON.errors.join("\n"))
+        )
+        .finally(() => {
+          this.send("closeModal");
+          bootbox.alert(I18n.t("discourse_subscriptions.admin.canceled"));
```
Comment (romanrizzi) [inline assets/javascripts/discourse/routes/admin-plugins-discourse-subscriptions-subscriptions.js.es6:23]: `bootbox#alert` will be called twitce if the request fails. Maybe this should be moved inside the `then` callback?

File: app/controllers/discourse_subscriptions/admin/subscriptions_controller.rb [hunk 2]
```diff
@@ -49,6 +51,18 @@ def destroy
...
+
+      private
+
+      # this will only refund the most recent subscription payment
+      def refund_subscription(subscription_id)
+        subscription = ::Stripe::Subscription.retrieve(subscription_id)
+        invoice = ::Stripe::Invoice.retrieve(subscription[:latest_invoice]) if subscription[:latest_invoice]
+        payment_intent = invoice[:payment_intent] if invoice[:payment_intent]
+        refund = ::Stripe::Refund.create({
+          payment_intent: payment_intent,
```
Comment (romanrizzi) [inline app/controllers/discourse_subscriptions/admin/subscriptions_controller.rb:63]: What happens if `payment_intent` is nil?

**Review by romanrizzi — CHANGES_REQUESTED** [review]

File: assets/javascripts/discourse/routes/admin-plugins-discourse-subscriptions-subscriptions.js.es6 (hunk 1 above)
Comment (justindirose) [reply assets/javascripts/discourse/routes/admin-plugins-discourse-subscriptions-subscriptions.js.es6:23]: Good call. Fixing that.

File: app/controllers/discourse_subscriptions/admin/subscriptions_controller.rb (hunk 2 above)
Comment (justindirose) [reply app/controllers/discourse_subscriptions/admin/subscriptions_controller.rb:63]: The call will fail which will recover an error to the client from the rescue on line 50. Though the likelihood of it getting into this state is very low as if an invoice exists, a payment intent exists.

Comment (romanrizzi) [reply app/controllers/discourse_subscriptions/admin/subscriptions_controller.rb:63]: Oops, I missed the rescue.

**Comment by justindirose** [comment]
@romanrizzi should be fixed up now. Can you review and approve changes if you see fit?

**Review by romanrizzi — APPROVED** [review]

**Pull request merged**